main.tf

```diff
@@ -143,10 +172,15 @@ resource "google_spanner_database_iam_member" "database" {
 }
 
 module "schedule_spanner_backup" {
-  count                  = length(local.backup_args) > 0 ? 1 : 0
+  for_each = { for idx, backup_arg in local.backup_args : idx => backup_arg }
+
   source                 = "./modules/schedule_spanner_backup"
   project_id             = var.project_id
-  backup_schedule        = var.backup_schedule
-  workflow_argument      = jsonencode(local.backup_args)
-  backup_schedule_region = var.backup_schedule_region
+  instance_name          = var.instance_name
+  database_name          = each.value.database
+  retention_duration     = each.value.expireTime
+  cron_spec_text         = var.backup_schedule
```

Comment:
Is this intended?  Perhaps should be `var.cron_spec_text`?  Otherwise that variable isn't currently used in the root module.